op-chain-ops/genesis/layer_two.go

```diff
@@ -40,10 +40,15 @@ func BuildL2Genesis(config *DeployConfig, dump *ForgeAllocs, l1StartBlock *types
 		return nil, fmt.Errorf("deploy config mismatch with allocs. Deploy config fundDevAccounts: %v, actual allocs: %v", config.FundDevAccounts, hasDevAccounts)
 	}
 	// sanity check the permit2 immutable, to verify we using the allocs for the right chain.
-	chainID := [32]byte(genspec.Alloc[predeploys.Permit2Addr].Code[6945 : 6945+32])
-	expected := uint256.MustFromBig(genspec.Config.ChainID).Bytes32()
-	if chainID != expected {
-		return nil, fmt.Errorf("allocs were generated for chain ID %x, but expected chain %x (%d)", chainID, expected, genspec.Config.ChainID)
+	if permit2 := genspec.Alloc[predeploys.Permit2Addr].Code; len(permit2) != 0 {
```

Comment:
We should add a check to enforce the `permit2 presence, not silently ignore it if it's missing.

See original issue, it appears like a L1 state dump was passed as a L2 allocs.

Reply:
This is fair, I don't love that we couple so tightly what the allocs should be in this way but its definitely an approach, there are some allocs that MUST exist for the system to work and permit2 is technically not one of them

Reply:
I'm fine with allowing a missing permit2 preinstall, if we at least keep some sanity checks that the allocs were properly configured. It's too easy to just accidentally point at a L1 chain genesis, or an allocs file of another chain / test, if it's completely silent.